src/core_plugins/vega/public/vega_request_handler.js

```diff
@@ -13,7 +13,8 @@ export function VegaRequestHandlerProvider(Private, es, timefilter, serviceSetti
 
     name: 'vega',
 
-    handler(vis) {
+    handler(vis, { timeRange }) {
+      timeCache.setTimeRange(timeRange);
```

Comment:
Pass in the `timeRange` that the vega request handler got into the timecache, that is used.